src/notebooks/controllers/notebookIPyWidgetCoordinator.ts

```diff
@@ -36,15 +36,15 @@ class NotebookCommunication implements IWebviewCommunication, IDisposable {
         this.changeController(controller);
     }
     public changeController(controller: IVSCodeNotebookController) {
-        if (this.controller === controller) {
+        if (this.controller.id === controller.id) {
```

Comment:
we have an id to uniquely identify controllers, hence using that when comparing controllers.